directembedding/src/main/scala/ch/epfl/directembedding/Utils.scala

```diff
@@ -87,3 +93,16 @@ trait DirectEmbeddingUtils extends MacroModule with TransformationUtils {
   }
 
 }
+
+trait TypeHelper extends MacroModule {
+  import c.universe._
+
+  private def name(s: Symbol): String = s.name.toString.trim
+
+  def equalSymbols(a: Symbol)(b: Symbol): Boolean =
```

Comment:
Feel free to take on this one, I won't be able to look into it until Thursday.

I don't know if it's necessary, but it may help to change the signature to include the trees.